src/CommunityToolkit.Maui.Core/Interfaces/IDockLayout.shared.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace CommunityToolkit.Maui.Core.Interfaces;
+
+/// <summary>Docking position for a view in <see cref="IDockLayout"/>.</summary>
+public enum DockEnum
+{
+	/// <summary>Do not dock.</summary>
+	None = 0,
```

Comment:
What's the purpose of `None`? Can we remove it?

Or could we rename it to `Unassigned` and then throw an `Exception` in `DockLayoutManager.Measure` if the dev hasn't specified a position?   

```cs
public override Size Measure(double widthConstraint, double heightConstraint)
{
  // ...

  switch (position)
  {
    case DockPosition.Unassigned:
throw new InvalidOperationException($"{position} invalid. {nameof(DockPosition)} must be assigned.");
    // ...
  }

  // ...
}

Reply:
_None_ is the default value. It essentially means, do not dock the view anywhere to the sides (top, left, right, bottom), but place it as 'content' (= center, if all other positions are occupied). I thought to make the default case explicit (also simplifies the logic a little bit).

Reply:
In what scenarios would a developer use a DockLayout with `DockLayout.None`?

Reply:
If you look at the sample page, the last child (which typically fills the remaining space) is specified with `DockPosition.None`. `DockPosition.None` is the same as omitted it (default value).
`None` works similar to `Top`, but here both width+height are considered. For `Top` and `Bottom` only the height is applied explicitely (width is given by the surrounding view proportions). For `Left`and `Right` its only the width.